### ISSUE TITLE
XML - <initializer> for array of structure only contains references in first array element

### ISSUE THREAD

**Issue by krausedj** [doxygen/doxygen#6950]
For C code, if you have a array of structures

```
struct S_TAG {
     func_ptr foo;
     uint8 data:
}:
typedef struct S_TAG S_TYPE;

S_TYPE data[3] = {
    { (func_ptr)foo, 0 },
    { (func_ptr)bar, 1 },
    { (func_ptr}dog, 2 }
};
```

There will only ever be references created for the first func_ptr, and foo, the other array elements will not output references.  In my test setup, the HTML contains links for foo, bar, and dog.  Regular arrays work fine.

Would the target debug entry point be the following?

```
    t << "        <initializer>";
    linkifyText(TextGeneratorXMLImpl(t),def,md->getBodyDef(),md,md->initializer());
    t << "</initializer>" << endl;
```

1.8.15

**Comment by albert-github** [comment]
Although an example is given (and a pointer for possible debug starting).
To be sure we are talking about the same: please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by krausedj** [comment]
[test_initializer.zip](https://github.com/doxygen/doxygen/files/3134510/test_initializer.zip)

Yes, that is fair, sorry it was late last night.  I was able to replicate it in a very small test case, which is attached.

You can see d5/d04/group___local_group_1ga5126e75ea4e84ac62fb05b1d3ec641b5 which shows the issue.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6951

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
